src/new-components/checkbox-field/checkbox-field.tsx

```diff
@@ -104,7 +106,10 @@ const CheckboxField = React.forwardRef<HTMLInputElement, CheckboxFieldProps>(fun
                 indeterminate={indeterminate}
                 aria-hidden
             />
-            {label ? <Text>{label}</Text> : null}
+            <Box display="flex" alignItems="center" className={styles.content}>
+                {icon ? <>{icon}</> : null}
+                {label ? <Text>{label}</Text> : null}
+            </Box>
```

Comment:
Thank you for pointing me to `Button`/`ButtonLink`, that was very helpful and I should've looked at that to begin with, as the use case is quite similar. In my current implementation I switched to 

```
type IconElement = React.ReactChild
```

which is what we use in `Button`/`ButtonLink`. Let me know if this is satisfactory.

While looking at that code, I also notice that we allow `startIcon` and `endIcon`.
Would it make sense to add `startIcon` and `endIcon` to `CheckboxField` (instead of just `icon`), even though we don't have (at least for the moment) any Design requests or any use case for an `endIcon`?

Reply:
Sure, whatever we use in `Button`, if it has been satisfactory for now, and if it covers this future use case of yours where you'll need to use `<img />`, then it's all good.